internal/topic/validation.go

```diff
@@ -37,6 +37,25 @@ func GetValidatedModel(ctx context.Context, req *metrov1.Topic) (*Model, error)
 	return m, nil
 }
 
+// GetValidatedTopicForAdminUpdate validates an incoming proto request and returns the model
+func GetValidatedTopicForAdminUpdate(ctx context.Context, req *metrov1.AdminTopic) (*Model, error) {
+	p, t, err := ExtractTopicMetaAndValidateForCreate(ctx, req.GetName())
+	if err != nil {
+		return nil, merror.Newf(merror.InvalidArgument, "Invalid [topics] name: (name=%s)", req.Name)
+	}
+	m := &Model{}
+	m.Name = req.GetName()
```

Comment:
this shouldn't be allowed for update ?

Reply:
Lookup on topic `name` will be done using this. If a wrong/different name is sent, say for the update, then this lookup would fail.